src/simulation/dynamics/constraintEffector/_UnitTest/test_ConstraintDynamicEffectorUnit.py

```diff
@@ -0,0 +1,331 @@
+# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+# ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+# OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+#
+#   Unit Test Script
+#   Module Name:        constraintEffector
+#   Author:             João Vaz Carneiro and Andrew Morell
+#   Creation Date:      November 8, 2022
+#
+
+import inspect
+import os
+import pytest
+import matplotlib.pyplot as plt
+import numpy as np
+
+from Basilisk.utilities import SimulationBaseClass, unitTestSupport, orbitalMotion, macros, RigidBodyKinematics
+from Basilisk.simulation import spacecraft, constraintDynamicEffector, gravityEffector, svIntegrators
+
+# uncomment this line if this test is to be skipped in the global unit test run, adjust message as needed
+# @pytest.mark.skipif(conditionstring)
+# uncomment this line if this test has an expected failure, adjust message as needed
+# @pytest.mark.xfail()
+
+@pytest.mark.parametrize("testCase", ["NoGravity","Gravity"])
+def test_constraintEffector(show_plots,testCase):
+    r"""Module Unit Test
+    **Validation Test Description**
+
+    This unit test sets up two spacecraft connected by a holonomic constraint effector acting as a physical connection
+    between them. The two spacecraft are set up with an identical mass and symmetrical inertias. One scenario includes
+    gravity to check orbital conservation while the other scenario excludes gravity to check more sensitive constraint
+    violations.
+
+    **Description of Variables Being Tested**
+
+    In this file we are checking the principles of conservation of energy and angular momentum. Both the orbital and
+    rotational energy and angular momentum must be maintained. While Baumgarte stabilization is an approximate method,
+    there is still an expected threshold below which these quantities are conserved:
+
+    - ``combinedOrbAngMom``
+    - ``combinedOrbEnergy``
+    - ``combinedRotAngMom``
+    - ``combinedRotEnergy``
+
+    We are also checking the constraint violation magnitudes throughout the simulation. Due to the nature of the 
+    Baumgarte stabilization method, there must first be a constraint violation before a force is applied to correct 
+    the constraint violation and keep the spacecraft aligned. For slow moving spacecraft systems such as this, the 
+    constraint violations are expected to remain below a threshold dependent on how dynamic a scenario is. The 
+    threshold for the scenario with gravity therefore expects a larger threshold than the scenario without gravity. 
+    The direction and attitude constraint violations checked are:
+
+    - ``psi_B1``
+    - ``sigma_B2B1``
+    
+    """
+
+    constraintEffector(show_plots,testCase)
+
+def constraintEffector(show_plots, testCase):
```

Comment:
It's good practice for unit tests to only test one thing. Here, the ``testCase`` variable is used to change the unit test. Why not separate this into two different tests? It might make for some repeat code, but it will make the unit test much more readable in the future. This is definitely not required, but think about it.

Reply:
Agreed, these are now two different tests